### DISCUSSION TITLE
Smart Meter Interface with Iskra MT176

### DISCUSSION THREAD

**Discussion by bgdev2015** [arendst/Tasmota#16189] [Q&A]
Is there a way to get the readings from an Iskra MT176? I havnt found a way for it.

**Comment by barbudor** [comment]
Is there any kind of interface to connect a tasmotized ESP?

May be this could give some hints
![image](https://user-images.githubusercontent.com/2996042/183745076-8b97b71f-bea5-4426-8066-6717275b8414.png)

**Reply by bgdev2015** [reply]
I know that the Smart Meter Interface can handle some Iskra Models, but the MT176 isnt listed here:
https://tasmota.github.io/docs/Smart-Meter-Interface/#driver-commands
Is it possible to get the readings from my MT176?

Its not the question how to get, its more like is it possible to get the Readings^^

**Reply by barbudor** [reply]
If there is an interface and if you have the protocol, then it is probably possible
but without any details, it's just a guessing game

**Reply by bgdev2015** [reply]
there is an Infared-Interface, like your Picture. But more i dont know, i havnt an ESP8266 by side atm. 
I asked because the MT176 isnt listed. 

Now i would buy an ESP8266 with an Infaredadapter and try it. If there any Problems i would ask again^^

**Reply by barbudor** [reply]
until you have the details of the protocol, you won't be able to go further
however, most of the type, the IR interface is just a pulse so it wouldn't provide any value reading, juste a frequency that you can convert into A or kWh

**Reply by sfromis** [reply]
What I see on my meter (not Iskra) is that the simple pulse count interface is not blinking where such a round probe is meant to be placed, but on separate points which looks more looks like what in that image is further down and a bit to the left of the cable.

Here's a brief intro article about the much more informational interface for the round IR pod:
https://intranet-of-things.com/smarthome/house/technic/electricmeter/

> Most modern electricity meters or smart meters (often on the digital display to detect) have an IR interface via which you can read the current consumption and other data. The data is read out with the help of an IR read head.

On Discord, nickctrl posted these specific relevant links:
https://wiki.volkszaehler.org/hardware/channels/meters/power/edl-ehz/iskraemeco_mt176
https://wiki.volkszaehler.org/software/sml?_x_tr_sl=auto&_x_tr_tl=fr&_x_tr_hl=en&_x_tr_pto=wapp

**Comment by joba-1** [comment]
according to this forum thread https://de.elv.com/forum/iskra-mt176-17126  it works like the 175. SML via IR serial 9600 baud. 
Exact settings here: https://files2.elv.com/public/14/1421/142148/Internet/142148_zaehlerliste_data.pdf

You don't need expensive ir readers (I have seen some for ridiculous ~100€) 
I just soldered a generic transistor and an ir led together. Cost: less than 100ct

Here are some more details of my project with an Itron

EDIT: and with „Here“ I meant this https://github.com/joba-1/ElectricityMeter :D

**Comment by bgdev2015** [comment]
I will get my ESP8266 with the IR-Modul next Month. Both from AZDelivery.
My complete set with cables are about 20€.

NodeMCU Lolin V3: https://www.amazon.de/gp/product/B06Y1ZPNMS/ref=ox_sc_saved_title_1?smid=A1X7QLRQH87QA3&psc=1
IR-Modul: https://www.amazon.de/gp/product/B07VP4JDTH/ref=ox_sc_saved_title_3?smid=A1X7QLRQH87QA3&psc=1
Additional Cables: https://www.amazon.de/gp/product/B074P726ZR/ref=ox_sc_saved_image_2?smid=A1X7QLRQH87QA3&psc=1

I think this should be working.
If iam ready to go and get some Problems ill get some quick response here^^

**Reply by Sfre12** [reply]
My suggestion is to take a IR Head like this.
It is already with magnet and I made already two times positive experience
https://www.ebay.de/itm/313884760667?hash=item4914fd5a5b:g:DFwAAOSwpZtgM9OW
From this seller is also a NON USB Version aviliable. check it

**Comment by bgdev2015** [comment]
Ok now i get all values but there also values that cant be correct. Like total-consumed thats more than million kw/h or an output without any PV or something that can produce.

**Comment by bgdev2015** [comment]
Someone here that can be help with an correct Script?
